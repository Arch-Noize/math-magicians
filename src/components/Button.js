import React from 'react';
import './styles.css';
import PropTypes from 'prop-types';

const Button = (props) => {
  const { number, handleEvent } = props;
  if (number === '\u00F7' || number === 'x' || number === '-' || number === '+' || number === '=') {
    return <button type="button" className="button" id="operator" onClick={handleEvent}>{number}</button>;
  } if (number === '0') {
    return <button type="button" className="button" id="zero" onClick={handleEvent}>{number}</button>;
  }
  return <button type="button" className="button" onClick={handleEvent}>{number}</button>;
};

Button.propTypes = {
  number: PropTypes.number.isRequired,
  handleEvent: PropTypes.func.isRequired,
};

export default Button;
