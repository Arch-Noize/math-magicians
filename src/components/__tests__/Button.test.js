import React from 'react';
import { render } from '@testing-library/react';
import Button from '../Button';

describe('Loading button', () => {
    it('renders correctly', () => {
        const handleEvent = jest.fn();
        const {tree} = render(<Button number={1} handleEvent={handleEvent}/>)
        expect(tree).toMatchSnapshot();   
    });
})