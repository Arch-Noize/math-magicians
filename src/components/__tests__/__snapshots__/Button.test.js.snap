// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading button renders correctly 1`] = `undefined`;
